src/plugins/application_config/server/routes/index.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  IRouter,
+  IScopedClusterClient,
+  Logger,
+  OpenSearchDashboardsResponseFactory,
+} from '../../../../core/server';
+import { ConfigurationClient } from '../types';
+
+export function defineRoutes(
+  router: IRouter,
+  getConfigurationClient: (inputOpenSearchClient: IScopedClusterClient) => ConfigurationClient,
+  logger: Logger
+) {
+  router.get(
```

Comment:
side note: i wonder why we dont have a router.head? Might have to look into this.

Reply:
I have updated the code to use `route.get/delete/post` to avoid putting the head names into API path. Let me know if this addressed your comment.